### ISSUE TITLE
Executables built with cx_freeze broken after On-host pixels to PDF conversion PR was merged

### ISSUE THREAD

**Issue by jkarasti** [freedomofpress/dangerzone#974]
### What happened?

On-host pixels to PDF conversion merged in #748 uncovered an incompatibility between pumypdf and cx_freeze which causes an error when running `dangerzone.exe` or `dangerzone-cli.exe`  built with cx_freeze.

### operating system version

Windows 10 Pro 22H2 (OS Build 19045.5073)

### Dangerzone version

Top of main branch

### Additional info

Here's the error log I got:

```
Traceback (most recent call last):
  File "C:\Users\juhok\AppData\Local\pypoetry\Cache\virtualenvs\dangerzone-_8ncJGJf-py3.12\Lib\site-packages\cx_Freeze\initscripts\__startup__.py", line 140, in run
    module_init.run(f"__main__{name}")
  File "C:\Users\juhok\AppData\Local\pypoetry\Cache\virtualenvs\dangerzone-_8ncJGJf-py3.12\Lib\site-packages\cx_Freeze\initscripts\console.py", line 25, in run
    exec(code, main_globals)
  File "install\windows\dangerzone-cli.py", line 1, in <module>
  File "C:\GitHub\dangerzone\dangerzone\__init__.py", line 27, in <module>
    from .cli import cli_main as main
  File "C:\GitHub\dangerzone\dangerzone\cli.py", line 10, in <module>
    from .isolation_provider.container import Container
  File "C:\GitHub\dangerzone\dangerzone\isolation_provider\container.py", line 12, in <module>
    from .base import IsolationProvider, terminate_process_group
  File "C:\GitHub\dangerzone\dangerzone\isolation_provider\base.py", line 12, in <module>
    import fitz
  File "C:\Users\juhok\AppData\Local\pypoetry\Cache\virtualenvs\dangerzone-_8ncJGJf-py3.12\Lib\site-packages\fitz\__init__.py", line 2, in <module>
    from pymupdf import *
  File "C:\Users\juhok\AppData\Local\pypoetry\Cache\virtualenvs\dangerzone-_8ncJGJf-py3.12\Lib\site-packages\pymupdf\__init__.py", line 21706, in <module>
    from . import utils
ImportError: cannot import name 'utils' from partially initialized module 'pymupdf' (most likely due to a circular import) (C \GitHub\dangerzone\build\exe.win-amd64-3.12\lib\pymupdf\__init__.pyc)
```

**Comment by almet** [comment]
Hey, thanks for the report. I'm curious: how did you run the `dangerzone.exe`, and did you update the installed dependencies? It seems weird that this error wasn't catched by the CI!

**Comment by jkarasti** [comment]
> Hey, thanks for the report. I'm curious: how did you run the `dangerzone.exe`

I've built the executables with `poetry run python .\setup-windows.py build` and then tried to launch dangerzone.exe inside `exe.win-amd64-3.12` but hit the above error. Interestingly enough, running Dangerzone with `poetry run .\dev_scripts\dangerzone.bat` instead of the executables produced by cx_freeze works a-okay.

> Did you update the installed dependencies?

I've tried to get Dangerzone running with a myriad of combinations of versions of cx_freeze between 7.2.0 and latest and pymupdf between 1.23.3 and latest. As far as I can tell, Dangerzone runs from commit 4abd4720 (commit before on-host PR) regradless of which version of cx_freeze or pymupdf is installed

>  It seems weird that this error wasn't catched by the CI!

At a cursory glance, I don't think CI runs the .exe that cx_freeze builds, so I don't think this particular edge case can get caught in the first place.
